src/components/scenes/RequestScene.js

```diff
@@ -443,46 +490,143 @@ export class Request extends React.Component<Props, State> {
+
+    const plugins: Object = SETTINGS_SELECTORS.getPlugins(state)
+    const allCurrencyInfos: EdgeCurrencyInfo[] = plugins.allCurrencyInfos
+    const currencyInfo: EdgeCurrencyInfo | void = getCurrencyInfo(allCurrencyInfos, currencyCode)
+
+    if (!guiWallet || !currencyCode) {
+      return {
+        currencyCode: null,
+        currencyInfo: null,
+        edgeWallet: null,
+        exchangeSecondaryToPrimaryRatio: null,
+        guiWallet: null,
+        loading: true,
+        primaryCurrencyInfo: null,
+        secondaryCurrencyInfo: null,
+        publicAddress: '',
+        legacyAddress: '',
+        useLegacyAddress: null,
+        fioAddressesExist: false,
+        isConnected: state.network.isConnected
+      }
+    }
+
+    const edgeWallet: EdgeCurrencyWallet = currencyWallets[guiWallet.id]
+    const primaryDisplayDenomination: GuiDenomination = SETTINGS_SELECTORS.getDisplayDenomination(state, currencyCode)
+    const primaryExchangeDenomination: GuiDenomination = UI_SELECTORS.getExchangeDenomination(state, currencyCode)
+    const secondaryExchangeDenomination: GuiDenomination = getDenomFromIsoCode(guiWallet.fiatCurrencyCode)
+    const secondaryDisplayDenomination: GuiDenomination = secondaryExchangeDenomination
+    const primaryExchangeCurrencyCode: string = primaryExchangeDenomination.name
+    const secondaryExchangeCurrencyCode: string = secondaryExchangeDenomination.name ? secondaryExchangeDenomination.name : ''
+
+    const primaryCurrencyInfo: GuiCurrencyInfo = {
+      displayCurrencyCode: currencyCode,
+      displayDenomination: primaryDisplayDenomination,
+      exchangeCurrencyCode: primaryExchangeCurrencyCode,
+      exchangeDenomination: primaryExchangeDenomination
+    }
+    const secondaryCurrencyInfo: GuiCurrencyInfo = {
+      displayCurrencyCode: guiWallet.fiatCurrencyCode,
+      displayDenomination: secondaryDisplayDenomination,
+      exchangeCurrencyCode: secondaryExchangeCurrencyCode,
+      exchangeDenomination: secondaryExchangeDenomination
+    }
+    const isoFiatCurrencyCode: string = guiWallet.isoFiatCurrencyCode
+    const exchangeSecondaryToPrimaryRatio = UI_SELECTORS.getExchangeRate(state, currencyCode, isoFiatCurrencyCode)
+    const fioAddressesExist = !!state.ui.scenes.fioAddress.fioAddresses.length
+
+    // balance
+    const isToken = guiWallet.currencyCode !== currencyCode
+    const nativeBalance = isToken ? guiWallet.nativeBalances[currencyCode] : guiWallet.primaryNativeBalance
+    const displayBalance = truncateDecimals(bns.div(nativeBalance, primaryDisplayDenomination.multiplier, DIVIDE_PRECISION), 6)
+    const balance = formatNumber(decimalOrZero(displayBalance, 6)) // check if infinitesimal (would display as zero), cut off trailing zeroes
+
+    return {
+      currencyCode,
+      currencyInfo: currencyInfo || null,
+      edgeWallet,
+      exchangeSecondaryToPrimaryRatio,
+      guiWallet,
+      publicAddress: guiWallet.receiveAddress.publicAddress || '',
```

Comment:
Make sure 'receiveAddress' is actually on guiWallet before using it.
And use '??' instead of '||'

And the same for the next line

Reply:
receiveAddress is mandatory on the guiWallet as per the types. But if you insist.

Reply:
oh yeah also publicAddress is also not optional. So that is

Reply:
So the correct way if we follow the types should be,

publicAddress: guiWallet.receiveAddress.publicAddress,
legacyAddress: guiWallet.receiveAddress.legacyAddress ?? ''

But just did the checks to make sure receiveAddress is on guiWallet as you say